tests/Unit/Visualization/Python/Test_InterpolateVolumeData.py

```diff
@@ -0,0 +1,192 @@
+                        Spectral.Quadrature.GaussLobatto,
+                        Spectral.Quadrature.GaussLobatto
+                    ]))
+
+            target_vol.write_volume_data(self.observation_ids[j],
+                                         self.observation_values[j],
+                                         volume_data)
+
+    def tearDown(self):
+        os.remove(self.file_name)
+        try:
+            os.remove(os.path.join(self.path, "interpolated_1.h5"))
+        except OSError:
+            pass
+
+    def check_interpolation(self, file_name, sol1, sol2):
+
+        file = spectre_h5.H5File(file_name, "r")
+
+        vol = file.get_vol("/VolumeDataInterpolated")
+
+        self.assertEqual(vol.get_dimension(), 3)
+
+        #python2 support...
+        try:
+            self.assertItemsEqual(vol.list_observation_ids(),
+                                  self.observation_ids)
+        except AttributeError:
+            self.assertCountEqual(vol.list_observation_ids(),
+                                  self.observation_ids)
+
+        for i, obs in enumerate(self.observation_ids):
+            self.assertEqual(vol.get_grid_names(obs), self.element_names)
+            self.assertEqual(vol.get_observation_value(obs),
+                             self.observation_values[i])
+            self.assertEqual(vol.list_tensor_components(obs),
+                             self.tensor_names)
+            self.assertEqual(vol.get_extents(obs), [[4, 4, 4], [4, 4, 4]])
+            self.assertEqual(vol.get_bases(obs), [["Legendre"] * 3] * 2)
+            self.assertEqual(vol.get_quadratures(obs), [["Gauss"] * 3] * 2)
+
+        tensor1_obs_1 = np.asarray(vol.get_tensor_component(42, 'tensor1'))
+        tensor2_obs_1 = np.asarray(vol.get_tensor_component(42, 'tensor2'))
+
+        tensor1_obs_2 = np.asarray(vol.get_tensor_component(1000, 'tensor1'))
+        tensor2_obs_2 = np.asarray(vol.get_tensor_component(1000, 'tensor2'))
+
+        file.close()
+
+        self.assertTrue(np.allclose(tensor1_obs_1[:64], sol1, 1e-7, 1e-7))
+        self.assertTrue(np.allclose(tensor1_obs_1[64:], sol1 / 2., 1e-7, 1e-7))
+        self.assertTrue(np.allclose(tensor2_obs_1[:64], sol2, 1e-7, 1e-7))
+        self.assertTrue(np.allclose(tensor2_obs_1[64:], sol2 / 2., 1e-7, 1e-7))
+
+        self.assertTrue(np.allclose(tensor1_obs_2[:64], sol1 / 4., 1e-7, 1e-7))
+        self.assertTrue(np.allclose(tensor1_obs_2[64:], sol1 / 8., 1e-7, 1e-7))
+        self.assertTrue(np.allclose(tensor2_obs_2[:64], sol2 / 4., 1e-7, 1e-7))
+        self.assertTrue(np.allclose(tensor2_obs_2[64:], sol2 / 8., 1e-7, 1e-7))
+
+    def test_simple_interpolation(self):
+
+        mesh = Spectral.Mesh3D([4, 4, 4], [Spectral.Basis.Legendre] * 3,
+                               [Spectral.Quadrature.Gauss] * 3)
+        InterpolateVolumeData.interpolate_h5_file(self.file_name, mesh)
+
+        self.check_interpolation(self.file_name, self.sol1, self.sol2)
+
+
```

Comment:
Some of the logic like filtering of tensor components and observations isn't tested. That's probably ok, though if you want to add tests you can move the relevant code into small functions and test those.